File: 백준/Silver/2559. 수열/수열.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        var in = new BufferedReader(new InputStreamReader(System.in));
        var out = new PrintWriter(System.out);
        StringTokenizer st;

        st = new StringTokenizer(in.readLine());
        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());

        int[] nums = new int[n + 1];
        st = new StringTokenizer(in.readLine());
        for(int i = 0; i < n; i++){
            nums[i] = Integer.parseInt(st.nextToken());
        }


        int sum = 0;
        for(int i = 0; i < k; i++){
            sum += nums[i];
        }
        int ret = sum;

        for(int i = k; i < n; i++){
            sum = sum - nums[i - k] + nums[i];
            ret = Math.max(ret, sum);
        }
        out.println(ret);

        out.flush();
        out.close();
    }
}

File: 백준/Silver/2559. 수열/수열.py
psum = [0]*100004

n, k = map(int, input().split())
nums = list(map(int, input().split()))
for i in range(1, len(nums)+1):
    psum[i] = psum[i-1] + nums[i-1]
ret = -1e9
for i in range(k, len(nums)+1):
    ret = max(ret, psum[i] - psum[i-k])
print(ret)